code-challenges.test.js

```diff
@@ -10,52 +10,172 @@
+      "Cherry",
+      "Blueberry",
+      "Peach",
+    ]);
+  });
+});
 
+// b) Create the function that makes the test pass.
 
+// create a function that takes two arguments, an array and a string of a single letter ✅
+// use .filter() to iterate through array to see if the word .includes() the letter while keeping casing in mind ✅
 
-// b) Create the function that makes the test pass.
+// const findWordsWithLetter = (arrayOfWords, stringLetter) => {
+//   return arrayOfWords.filter((word) =>
+//     word.toLowerCase().includes(stringLetter)
+//   );
+// };
 
+const findWordsWithLetter = (arrayOfWords, stringLetter) => {
+  if (stringLetter.length == 1) {
+    return arrayOfWords.filter((word) =>
+      word.toLowerCase().includes(stringLetter)
+    );
+  } else {
+    return "Make sure to only use a single letter as the second argument.";
+  }
+};
 
+// For the refactor I put a condition to check if the user put in a single letter or not
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-
-
+describe("fullHouseOrNot", () => {
+  const hand1 = [5, 5, 5, 3, 3];
+  // Expected output: true
+  const hand2 = [5, 5, 3, 3, 4];
+  // Expected output: false
+  const hand3 = [5, 5, 5, 5, 4];
+  // Expected output: false
+  it("takes in an array of 5 numbers and determines whether or not the array is a “full house”", () => {
+    expect(fullHouseOrNot(hand1)).toEqual(true);
+    expect(fullHouseOrNot(hand2)).toEqual(false);
+    expect(fullHouseOrNot(hand3)).toEqual(false);
+  });
+});
 
 // b) Create the function that makes the test pass.
+
+// create a function that takes in an array of numbers ✅
+// check if there are only two different kinds of numbers in the array with new Set() to remove duplicates and .length to check ✅
+// check if one of the numbers is repeated only 2x or 3x ✅
+// return true if it is, false if not ✅
+
+// const fullHouseOrNot = (arrayOfNums) => {
+//   if ([...new Set(arrayOfNums)].length == 2) {
+//     let count = 0;
+//     for (let i = 0; i < arrayOfNums.length; i++) {
+//       if (arrayOfNums[0] === arrayOfNums[i]) {
+//         count += 1;
+//       }
+//     }
+//     return count == 2 || count == 3;
+//   } else {
+//     return false;
+//   }
+// };
+
+const fullHouseOrNot = (arrayOfNums) => {
+  let occurances = 0;
+  if ([...new Set(arrayOfNums)].length == 2 && arrayOfNums.length === 5) {
+    arrayOfNums.forEach((num) =>
+      arrayOfNums[0] === num ? (occurances += 1) : occurances
+    );
+    return occurances == 2 || occurances == 3;
+  } else {
+    return false;
+  }
+};
+
```

Comment:
I like this approach. `.reduce` may be an option but I imagine you would use it in replace of `Set` and in this case I think `Set` gets you what you need with less code.